### PR TITLE
gists endpoint

### PR DESCRIPTION
**Pull request by brunodmsi**
Since it has a repo counter, why not a gist counter also?

### CODE REVIEW

**Comment by brunodmsi** [comment]
Removing yarn.lock since I tried to use it, but cancelled after I've seen that the project was not using it. And I forgot to remove.

File: server.js [hunk 1]
```diff
@@ -56,6 +56,25 @@ app.get('/repos/:user', async (req, res) => {
...
+	if (response.length === 0)
+		return res.status(404).send(response);
+
+	const badge = await request(
+		`https://img.shields.io/badge/Gists-${response.length}-brightgreen${req.originalUrl.slice(req.originalUrl.indexOf('?'))}`)
+		.raw();
+
+	res.contentType('image/svg+xml')
+		.header('Cache-Control', 'no-age', 'max-age=600')
+		.send(badge);
```
Comment (thinkverse) [inline server.js:75]: No need to store the badge in a constant, you can just follow what's been done in the other endpoints.

```suggestion
		.send(await request(
		`https://img.shields.io/badge/Gists-${response.length}-brightgreen${req.originalUrl.slice(req.originalUrl.indexOf('?'))}`)
		.raw());
```

File: server.js [hunk 2]
```diff
@@ -56,6 +56,25 @@ app.get('/repos/:user', async (req, res) => {
...
+	const response = await request(
+		`https://api.github.com/users/${user}/gists`
+	).header(githubHeaders).json();
+
+	if (response.length === 0)
+		return res.status(404).send(response);
+
+	const badge = await request(
+		`https://img.shields.io/badge/Gists-${response.length}-brightgreen${req.originalUrl.slice(req.originalUrl.indexOf('?'))}`)
+		.raw();
```
Comment (thinkverse) [inline server.js:71]: No need for this constant, refer to the below comment. 👍
```suggestion
```

File: server.js [hunk 3]
```diff
@@ -56,6 +56,25 @@ app.get('/repos/:user', async (req, res) => {
 		.send(await request(`https://img.shields.io/badge/Repos-${response.public_repos}-brightgreen${req.originalUrl.slice(req.originalUrl.indexOf('?'))}`).raw())
 })
 
+app.get('/gists/:user', async (req, res) => {
+	const { user } = req.params;
+
+	const response = await request(
```
Comment (puf17640) [inline server.js:62]: formatting is off

File: server.js [hunk 4]
```diff
@@ -56,6 +56,25 @@ app.get('/repos/:user', async (req, res) => {
...
+		`https://api.github.com/users/${user}/gists`
+	).header(githubHeaders).json();
+
+	if (response.length === 0)
```
Comment (puf17640) [inline server.js:66]: if you search for a user that does not exist, github returns this
```
{
    "message": "Not Found",
    "documentation_url": "https://docs.github.com/rest/reference/gists#list-gists-for-a-user"
}
```

checking for length only is not the way to go i guess

**Pull request merged**